### ISSUE TITLE
gerrit url format support

### ISSUE THREAD

**Issue by tulequ** [loft-sh/devpod#815]
**What happened?**  
Branch regex not working with gerrit
In Gerrit: url in format https://gerrit.googlesource.com/k8s-gerrit

Which is not match regex if we want to specify branch
```
branchRegEx      = regexp.MustCompile(`^([^@]*(?:git@)?[^@/]+/[^@/]+/[^@/]+)@([a-zA-Z0-9\./\-\_]+)$`)
```

**How can we reproduce the bug?** (as minimally and precisely as possible)  

```
devpod up https://gerrit.googlesource.com/k8s-gerrit@example
16:15:07 error Try using the --debug flag to see a more verbose output
16:15:07 fatal https://gerrit.googlesource.com/k8s-gerrit@example is neither a local folder, git repository or docker image
```

**Comment by tulequ** [comment]
another problem is username in gerrit is real username (not `git`), so `git@` will not match

**Comment by pascalbreuninger** [comment]
@tulequ thanks for opening the issue. To be honest, I have never even heard of gerrit until now. 
The branch problem should be fixed with #819 and you can start it like so:
`devpod up $WORKSPACE_NAME --source=git:https://gerrit.googlesource.com/k8s-gerrit@example` 
This will force the cli to treat it as git repository and afterwards, both commit and branch checkouts should work (after the next patch release)

How do PRs work in gerrit? Do you need to link to them as well?

**Comment by tulequ** [comment]
Example gerrit patchset (~PR) in gerrit (you can click **Download** button for guide)
https://chromium-review.googlesource.com/c/chromium/src/+/5056576

But I don't intent to use with PR, only branch is need

**Comment by pascalbreuninger** [comment]
the fix landed in `v0.5.0-alpha.0`, closing this issue. 

Let us know if you encounter more problems with Gerrit!
